### PR TITLE
LPS-187081|master

### PR DESCRIPTION
**Pull request by mateussandes23**
https://liferay.atlassian.net/browse/LPS-187081

### CODE REVIEW

**Comment by mateussandes23** [comment]
ci:test:sf

**Comment by mateussandes23** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 1]
```diff
@@ -1911,6 +1911,48 @@ definition {
 		}
 	}
 
+	@description = "LPS-187081 - Verify the user cannot delete the only custom field in an object definition"
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1914]: This should be the story ticket, not the automation ticket

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 2]
```diff
@@ -1911,6 +1911,48 @@ definition {
...
+	@priority = 5
+	test CannotDeleteSoleCustomField {
+		
+		task ("Given a custom object published with a custom field") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1918]: `Given a custom object with a custom field`

I put the wrong text on the ticket, that's my bad

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 3]
```diff
@@ -1911,6 +1911,48 @@ definition {
...
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object");
+
+			ObjectAdmin.goToFieldsTab();
+
+			ObjectAdmin.addObjectFieldViaUI(
+				enableShowFiles = "true",
+				fieldAttachment = "Upload Directly from the User's Computer",
+				fieldLabel = "Custom Attachment",
+				fieldType = "Attachment");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1934]: Can you add the field using the API instead? This test doesn't require us to add the field using the UI

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 4]
```diff
@@ -1911,6 +1911,48 @@ definition {
...
+			ObjectAdmin.goToFieldsTab();
+
+			ObjectAdmin.selectKebabMenuOption(kebabOption = "Delete");
+		}
+
+		task ("Then a message is seen stating the custom field cannot be deleted and the custom field isn't deleted") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "Modal#BODY",
+				value1 = "\"Custom Attachment\" is the only field of the published object definition and cannot be deleted.");
+		}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1953]: Like the text states, you need to assert the field itself isn't deleted. You need to include an assertion that the field is present on the page after asserting the text on the modal.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 5]
```diff
@@ -1911,6 +1911,48 @@ definition {
 		}
 	}
 
+	@description = "LPS-187081 - Verify the user cannot delete the only custom field in an object definition"
+	@priority = 5
+	test CannotDeleteSoleCustomField {
+
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1917]: You didn't include the property to add this to acceptance

**Review by timpak — CHANGES_REQUESTED** [review]
@mateussandes23 please see my comments below. You also need to fix the SF error as well, please make sure you fix any SF errors before submitting your PRs for review. Don't make a separate SF commit, fix the error in the commit that caused it.

**Comment by mateussandes23** [comment]
ci:test:sf

**Comment by timpak** [comment]
@mateussandes23 is this ready to be reviewed?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 6]
```diff
@@ -1911,6 +1911,59 @@ definition {
 		}
 	}
 
+	@description = "LPS-179803 - Verify the user cannot delete the only custom field in an object definition"
+	@priority = 5
+	test CannotDeleteSoleCustomField {
+		property custom.properties = "feature.flag.LPS-179803=true";
+
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1918]: You're still missing the property to add this to acceptance. Please see other priority 5 tests, like `CannotEditERCField` which include this property

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 7]
```diff
@@ -1911,6 +1911,59 @@ definition {
...
+			AssertTextEquals.assertPartialText(
+				locator1 = "Modal#BODY",
+				value1 = "\"Custom Attachment\" is the only field of the published object definition and cannot be deleted.");
+
+			Button.click(button = "Done");
+
+			AssertElementPresent(
+				key_name = "Custom Attachment",
+				locator1 = "ObjectAdmin#TABLE_LIST_TITLE",
+				value1 = ${key_name});
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1963]: You forgot to include the value here

**Review by timpak — CHANGES_REQUESTED** [review]
@mateussandes23 please see my comments below. Please read my review comments carefully because you forgot to include the acceptance property again. 

This isn't the first time I had to push back a PR twice for corrections, please pay closer attention in the future.

**Review by timpak — APPROVED** [review]

**Comment by timpak** [comment]
Hey @mateussandes23 can you create a ticket to remove the feature flag from this once it's merged?

**Comment by timpak** [comment]
ci:forward

**Comment by andreefarias** [comment]
ci:forward

**Comment by andreefarias** [comment]
ci:forward

**Comment by andreefarias** [comment]
ci:forward

**Comment by andreefarias** [comment]
ci:forward

**Comment by andreefarias** [comment]
ci:forward

**Comment by andreefarias** [comment]
ci:forward

**Comment by andreefarias** [comment]
ci:test:stable

**Comment by andreefarias** [comment]
ci:forward

**Comment by andreefarias** [comment]
ci:forward

**Comment by andreefarias** [comment]
ci:forward

**Comment by timpak** [comment]
ci:forward:force

**Comment by andreefarias** [comment]
I'll close this PR. @mateussandes23 send directly to Brian, please.

**Pull request closed without merging**